ember-primitives/src/components/menu.gts

```diff
@@ -0,0 +1,233 @@
+import { hash, uniqueId } from '@ember/helper';
+import { on } from '@ember/modifier';
+
+import { modifier } from 'ember-modifier';
+import { cell } from 'ember-resources';
+import { getTabster, getTabsterAttribute, Types } from 'tabster';
+
+import { Popover, type Signature as PopoverSignature } from './popover.gts';
+
+import type { TOC } from '@ember/component/template-only';
+import type { WithBoundArgs } from '@glint/template';
+
+type Cell<V> = ReturnType<typeof cell<V>>;
+
+const TABSTER_CONFIG_CONTENT = getTabsterAttribute(
+  {
+    mover: {
+      direction: Types.MoverDirections.Both,
+      cyclic: true,
+    },
+    deloser: {},
+  },
+  true
+);
+
+const TABSTER_CONFIG_TRIGGER = getTabsterAttribute(
+  {
+    deloser: {},
+  },
+  true
+);
+
+export interface Signature {
+  Args: PopoverSignature['Args'];
+  Blocks: {
+    default: [
+      {
+        arrow: PopoverSignature['Blocks']['default'][0]['arrow'];
+        Trigger: WithBoundArgs<typeof Trigger, 'triggerElement' | 'contentId' | 'isOpen' | 'hook'>;
+        Content: WithBoundArgs<
+          typeof Content,
+          'triggerElement' | 'contentId' | 'isOpen' | 'PopoverContent'
+        >;
+        isOpen: boolean;
+      },
+    ];
+  };
+}
+
+const Separator: TOC<{
+  Element: HTMLDivElement;
+  Args: {};
+  Blocks: { default: [] };
+}> = <template>
+  <div role="separator" ...attributes>
+    {{yield}}
+  </div>
+</template>;
+
+/**
+ * We focus items on `pointerMove` to achieve the following:
+ *
+ * - Mouse over an item (it focuses)
+ * - Leave mouse where it is and use keyboard to focus a different item
+ * - Wiggle mouse without it leaving previously focused item
+ * - Previously focused item should re-focus
+ *
+ * If we used `mouseOver`/`mouseEnter` it would not re-focus when the mouse
+ * wiggles. This is to match native menu implementation.
+ */
+function focusOnHover(e: PointerEvent) {
+  const item = e.currentTarget;
+
+  if (item instanceof HTMLElement) {
+    item?.focus();
+  }
+}
+
+const Item: TOC<{
+  Element: HTMLButtonElement;
+  Args: { onSelect?: (event: Event) => void };
+  Blocks: { default: [] };
+}> = <template>
+  <button
+    type="button"
+    role="menuitem"
+    {{! @glint-ignore !}}
+    {{(if @onSelect (modifier on "click" @onSelect))}}
+    {{on "pointermove" focusOnHover}}
+    ...attributes
+  >
+    {{yield}}
+  </button>
+</template>;
+
+const installContent = modifier<{
+  Element: HTMLElement;
+  Args: {
+    Named: {
+      isOpen: Cell<boolean>;
+      triggerElement: Cell<HTMLElement>;
+    };
+  };
+}>((element, _: [], { isOpen, triggerElement }) => {
+  // focus first focusable element on the content
+  const tabster = getTabster(window);
+  const firstFocusable = tabster?.focusable.findFirst({
+    container: element,
+  });
+
+  firstFocusable?.focus();
+
+  // listen for "outside" clicks
+  function onDocumentClick(e: MouseEvent) {
```

Comment:
I'm surprised they/tabster don't have a utility for this!

Reply:
Yeah, I didn't find anything for this.